pkg/linkcache/devices_linux.go

```diff
@@ -41,36 +42,37 @@ func TestNodeWithVolumes(volumes []string) *v1.Node {
 	}
 }
 
-func newDeviceCacheForNode(period time.Duration, node *v1.Node) *DeviceCache {
+func newDeviceCacheForNode(period time.Duration, node *v1.Node, driverName string, deviceUtils deviceutils.DeviceUtils) *DeviceCache {
 	deviceCache := &DeviceCache{
-		volumes: make(map[string]deviceMapping),
-		period:  period,
-		dir:     byIdDir,
+		symlinks:    make(map[string]deviceMapping),
+		period:      period,
+		deviceUtils: deviceUtils,
+		dir:         byIdDir,
 	}
 
 	// Look at the status.volumesInUse field.  For each, take the last section
 	// of the string (after the last "/") and call AddVolume for that
 	for _, volume := range node.Status.VolumesInUse {
-		klog.Infof("Adding volume %s to cache", string(volume))
-		vID, err := pvNameFromVolumeID(string(volume))
-		if err != nil {
-			klog.Warningf("failure to retrieve name, skipping volume %q: %v", string(volume), err)
+		volumeName := string(volume)
+		tokens := strings.Split(volumeName, "^")
+		if len(tokens) != 2 {
+			klog.V(5).Infof("Skipping volume %q because splitting volumeName on `^` returns %d tokens, expected 2", volumeName, len(tokens))
+			continue
+		}
+
+		// The first token is of the form "kubernetes.io/csi/<driver-name>" or just "<driver-name>".
+		// We should check if it contains the driver name we are interested in.
+		if !strings.Contains(tokens[0], driverName) {
+			klog.V(5).Infof("Skipping volume %q because it is not a %s volume.", volumeName, driverName)
```

Comment:
This will be common if a customer has multiple csi drivers. We probably don't need this log.